nebula_decoders/include/nebula_decoders/nebula_decoders_aeva/aeva_aeries2_decoder.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <nebula_common/aeva/packet_types.hpp>
+#include <nebula_common/aeva/point_types.hpp>
+#include <nebula_common/nebula_common.hpp>
+
+#include <pcl/point_cloud.h>
+#include <sys/types.h>
+
+#include <atomic>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <mutex>
+
+namespace nebula::drivers
+{
+
+class AevaAeries2Decoder
+{
+public:
+  using AevaPoint = aeva::PointXYZVIRCAEDT;
+  using AevaPointCloud = pcl::PointCloud<AevaPoint>;
+  using AevaPointCloudUniquePtr = std::unique_ptr<AevaPointCloud>;
+
+  using callback_t = std::function<void(AevaPointCloudUniquePtr, uint64_t)>;
+
+  AevaAeries2Decoder() : cloud_state_({std::make_unique<AevaPointCloud>(), 0}) {}
+
+  void processPointcloudMessage(const aeva::PointCloudMessage & message);
+
+  void registerPointCloudCallback(callback_t callback);
+
+  void onParameterChange(ReturnMode return_mode);
+
+private:
+  struct DecoderState
+  {
+    int32_t new_frame_index;
+    uint64_t time_per_marker_point_ns;
+    size_t line_index;
+    size_t point_index;
+    uint64_t absolute_time_ns;
+  };
+
+  struct PointcloudState
+  {
+    std::unique_ptr<AevaPointCloud> cloud;
+    uint64_t timestamp;
+  };
+
+  ReturnType getReturnType(uint32_t peak_id);
+
+  callback_t callback_;
+  std::atomic<ReturnMode> return_mode_;
```

Comment:
Yes, it can be updated during runtime on parameter change here: [return_mode_.store(return_mode);](https://github.com/mojomex/nebula/blob/a3826110cfd92d00a197efef1a3846e8984e68ff/nebula_decoders/src/nebula_decoders_aeva/aeva_aeries2_decoder.cpp#L92)

and is read from another thread here: [switch (return_mode_.load()) {](https://github.com/mojomex/nebula/blob/a3826110cfd92d00a197efef1a3846e8984e68ff/nebula_decoders/src/nebula_decoders_aeva/aeva_aeries2_decoder.cpp#L78)